lapce-data/src/db.rs

```diff
@@ -428,6 +430,46 @@ impl LapceDb {
         Ok(info)
     }
 
+    /// fetches all the files stored in the database that are tagged as `unsaved_buffer`.<br>
+    /// Returns a hashmap of the form HashMap<path, file_content><br>
+    /// *Note: The file is deleted from the database right after as to prevent "ghost" buffers*
+    pub fn get_unsaved_buffers(&self) -> Result<im::HashMap<String, String>> {
+        let sled_db = self.get_db()?;
+        let mut buffers = im::HashMap::new();
+        let mut unsaved_paths = Vec::new();
+
+        for val in sled_db.iter().keys() {
+            let val = val.unwrap();
+            let s = String::from_utf8((&val).to_vec())
+                .expect("invalid utf-8 sequence retrieving unsaved buffer");
+
+            if s.contains("unsaved_buffer") {
```

Comment:
This could perhaps be `s.starts_with("unsaved_buffer:")` to avoid grabbing anything that just happens to be named `unsaved_buffer`.